mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java

```diff
@@ -35,11 +36,12 @@ public static Builder builder() {
         return new Builder();
     }
 
-    public static Attachment from(byte[] bytes, String type) {
+    public static Attachment from(byte[] bytes, String type, Optional<String> name) {
         return builder()
                 .attachmentId(AttachmentId.forPayload(bytes))
```

Comment:
you should not have to call it from the outside if it can be done based on bytes in build() method